File: address-validation-canada-2-dot-net/REST/AVCA2Response.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace address_validation_canada_2_dot_net.REST
{
    public class AVCA2Response
    {
        public CanadianAddressInfoV2 CanadianAddressInfoV2 { get; set; }
        public CanadianAddressInfo CanadianAddressInfo { get; set; }
        public Error Error { get; set; }
        public override string ToString()
        {
            return $"AVCA2_Response: [CanadianAddressInfoV2={CanadianAddressInfoV2}, Error={Error}]";
        }
    }
    public class CanadianAddressInfoV2
    {
        public string Address { get; set; }
        public string Address2 { get; set; }
        public string Municipality { get; set; }
        public string Province { get; set; }
        public string PostalCode { get; set; }
        public string TimeZone { get; set; }
        public string AddressNumberFragment { get; set; }
        public string StreetNameFragment { get; set; }
        public string StreetTypeFragment { get; set; }
        public string DirectionalCodeFragment { get; set; }
        public string UnitTypeFragment { get; set; }
        public string UnitNumberFragment { get; set; }
        public string IsPOBox { get; set; }
        public string BoxNumberFragment { get; set; }
        public string StationInfo { get; set; }
        public string DeliveryMode { get; set; }
        public string DeliveryInstallation { get; set; }
        public string Corrections { get; set; }
        public string CorrectionsDescriptions { get; set; }
        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.AppendLine($"Address: {Address}");
            sb.AppendLine($"Address2: {Address2}");
            sb.AppendLine($"Municipality: {Municipality}");
            sb.AppendLine($"Province: {Province}");
            sb.AppendLine($"PostalCode: {PostalCode}");
            sb.AppendLine($"TimeZone: {TimeZone}");
            sb.AppendLine($"AddressNumberFragment: {AddressNumberFragment}");
            sb.AppendLine($"StreetNameFragment: {StreetNameFragment}");
            sb.AppendLine($"StreetTypeFragment: {StreetTypeFragment}");
            sb.AppendLine($"DirectionalCodeFragment: {DirectionalCodeFragment}");
            sb.AppendLine($"UnitTypeFragment: {UnitTypeFragment}");
            sb.AppendLine($"UnitNumberFragment: {UnitNumberFragment}");
            sb.AppendLine($"IsPOBox: {IsPOBox}");
            sb.AppendLine($"BoxNumberFragment: {BoxNumberFragment}");
            sb.AppendLine($"StationInfo: {StationInfo}");
            sb.AppendLine($"DeliveryMode: {DeliveryMode}");
            sb.AppendLine($"DeliveryInstallation: {DeliveryInstallation}");
            sb.AppendLine($"Corrections: {Corrections}");
            sb.AppendLine($"CorrectionsDescriptions: {CorrectionsDescriptions}");

            return sb.ToString();
        }
    }
    public class Error
    {
        public string Type { get; set; }
        public string TypeCode { get; set; }
        public string Desc { get; set; }
        public string DescCode { get; set; }
        public override string ToString()
        {
            return $"Type: {Type} " +
                $"TypeCode: {TypeCode} " +
                $"Desc: {Desc} " +
                $"DescCode: {DescCode} ";
        }
    }
    public class CanadianAddressInfo
    {
        public string Address { get; set; }
        public string Address2 { get; set; }
        public string Municipality { get; set; }
        public string Province { get; set; }
        public string PostalCode { get; set; }
        public string TimeZone { get; set; }
        public string AddressNumberFragment { get; set; }
        public string StreetNameFragment { get; set; }
        public string StreetTypeFragment { get; set; }
        public string DirectionalCodeFragment { get; set; }
        public string UnitTypeFragment { get; set; }
        public string UnitNumberFragment { get; set; }
        public string IsPOBox { get; set; }
        public string BoxNumberFragment { get; set; }
        public string StationInfo { get; set; }
        public string DeliveryMode { get; set; }
        public string DeliveryInstallation { get; set; }

        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.AppendLine($"Address: {Address}");
            sb.AppendLine($"Address2: {Address2}");
            sb.AppendLine($"Municipality: {Municipality}");
            sb.AppendLine($"Province: {Province}");
            sb.AppendLine($"PostalCode: {PostalCode}");
            sb.AppendLine($"TimeZone: {TimeZone}");
            sb.AppendLine($"AddressNumberFragment: {AddressNumberFragment}");
            sb.AppendLine($"StreetNameFragment: {StreetNameFragment}");
            sb.AppendLine($"StreetTypeFragment: {StreetTypeFragment}");
            sb.AppendLine($"DirectionalCodeFragment: {DirectionalCodeFragment}");
            sb.AppendLine($"UnitTypeFragment: {UnitTypeFragment}");
            sb.AppendLine($"UnitNumberFragment: {UnitNumberFragment}");
            sb.AppendLine($"IsPOBox: {IsPOBox}");
            sb.AppendLine($"BoxNumberFragment: {BoxNumberFragment}");
            sb.AppendLine($"StationInfo: {StationInfo}");
            sb.AppendLine($"DeliveryMode: {DeliveryMode}");
            sb.AppendLine($"DeliveryInstallation: {DeliveryInstallation}");
            return sb.ToString();
        }
    }


}

File: address-validation-canada-2-nodejs/REST/avca2_response.js
export class AVCA2Response {
    constructor(data = {}) {
        this.CanadianAddressInfoV2 = data.CanadianAddressInfoV2 ? new CanadianAddressInfoV2(data.CanadianAddressInfoV2) : null;
        this.CanadianAddressInfo = data.CanadianAddressInfo ? new CanadianAddressInfo(data.CanadianAddressInfo) : null;
        this.Error = data.Error ? new Error(data.Error) : null;
        this.Debug = data.Debug || [];
    }

    toString() {
        const debugString = this.Debug.length ? this.Debug.join(', ') : 'null';
        return `AVCA2Response: CanadianAddressInfoV2 = ${this.CanadianAddressInfoV2 ? this.CanadianAddressInfoV2.toString() : 'null'}, CanadianAddressInfo = ${this.CanadianAddressInfo ? this.CanadianAddressInfo.toString() : 'null'}, Error = ${this.Error ? this.Error.toString() : 'null'}, Debug = [${debugString}]`;
    }
}

export class CanadianAddressInfoV2 {
    constructor(data = {}) {
        this.Address = data.Address;
        this.Address2 = data.Address2;
        this.Municipality = data.Municipality;
        this.Province = data.Province;
        this.PostalCode = data.PostalCode;
        this.TimeZone = data.TimeZone;
        this.AddressNumberFragment = data.AddressNumberFragment;
        this.StreetNameFragment = data.StreetNameFragment;
        this.StreetTypeFragment = data.StreetTypeFragment;
        this.DirectionalCodeFragment = data.DirectionalCodeFragment;
        this.UnitTypeFragment = data.UnitTypeFragment;
        this.UnitNumberFragment = data.UnitNumberFragment;
        this.IsPOBox = data.IsPOBox;
        this.BoxNumberFragment = data.BoxNumberFragment;
        this.StationInfo = data.StationInfo;
        this.DeliveryMode = data.DeliveryMode;
        this.DeliveryInstallation = data.DeliveryInstallation;
        this.Corrections = data.Corrections;
        this.CorrectionsDescriptions = data.CorrectionsDescriptions;
    }

    toString() {
        return `CanadianAddressInfoV2: Address = ${this.Address}, Address2 = ${this.Address2}, Municipality = ${this.Municipality}, Province = ${this.Province}, PostalCode = ${this.PostalCode}, TimeZone = ${this.TimeZone}, AddressNumberFragment = ${this.AddressNumberFragment}, StreetNameFragment = ${this.StreetNameFragment}, StreetTypeFragment = ${this.StreetTypeFragment}, DirectionalCodeFragment = ${this.DirectionalCodeFragment}, UnitTypeFragment = ${this.UnitTypeFragment}, UnitNumberFragment = ${this.UnitNumberFragment}, IsPOBox = ${this.IsPOBox}, BoxNumberFragment = ${this.BoxNumberFragment}, StationInfo = ${this.StationInfo}, DeliveryMode = ${this.DeliveryMode}, DeliveryInstallation = ${this.DeliveryInstallation}, Corrections = ${this.Corrections}, CorrectionsDescriptions = ${this.CorrectionsDescriptions}`;
    }
}

export class Error {
    constructor(data = {}) {
        this.Type = data.Type;
        this.TypeCode = data.TypeCode;
        this.Desc = data.Desc;
        this.DescCode = data.DescCode;
    }

    toString() {
        return `Error: Type = ${this.Type}, TypeCode = ${this.TypeCode}, Desc = ${this.Desc}, DescCode = ${this.DescCode}`;
    }
}
export class CanadianAddressInfo {
    constructor(data = {}) {
        this.Address = data.Address;
        this.Address2 = data.Address2;
        this.Municipality = data.Municipality;
        this.Province = data.Province;
        this.PostalCode = data.PostalCode;
        this.TimeZone = data.TimeZone;
        this.AddressNumberFragment = data.AddressNumberFragment;
        this.StreetNameFragment = data.StreetNameFragment;
        this.StreetTypeFragment = data.StreetTypeFragment;
        this.DirectionalCodeFragment = data.DirectionalCodeFragment;
        this.UnitTypeFragment = data.UnitTypeFragment;
        this.UnitNumberFragment = data.UnitNumberFragment;
        this.IsPOBox = data.IsPOBox;
        this.BoxNumberFragment = data.BoxNumberFragment;
        this.StationInfo = data.StationInfo;
        this.DeliveryMode = data.DeliveryMode;
        this.DeliveryInstallation = data.DeliveryInstallation;
    }

    toString() {
        return `CanadianAddressInfo: Address = ${this.Address}, Address2 = ${this.Address2}, Municipality = ${this.Municipality}, Province = ${this.Province}, PostalCode = ${this.PostalCode}, TimeZone = ${this.TimeZone}, AddressNumberFragment = ${this.AddressNumberFragment}, StreetNameFragment = ${this.StreetNameFragment}, StreetTypeFragment = ${this.StreetTypeFragment}, DirectionalCodeFragment = ${this.DirectionalCodeFragment}, UnitTypeFragment = ${this.UnitTypeFragment}, UnitNumberFragment = ${this.UnitNumberFragment}, IsPOBox = ${this.IsPOBox}, BoxNumberFragment = ${this.BoxNumberFragment}, StationInfo = ${this.StationInfo}, DeliveryMode = ${this.DeliveryMode}, DeliveryInstallation = ${this.DeliveryInstallation}`;
    }
}

export default AVCA2Response;